### PR TITLE
Simplified `BaseSQLTableQueryEngine._parse_response_to_sql` and updated `DEFAULT_TEXT_TO_SQL_TMPL`

### PR DESCRIPTION
**Pull request by jamesbraza**
# Description

Fixes https://github.com/jerryjliu/llama_index/issues/7865
- Note: `lstrip` is a regex, not a prefix.  We need to use `removeprefix`
- Also fixed a broken URL

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

File: llama_index/indices/struct_store/sql_query.py [hunk 1]
```diff
@@ -243,17 +243,13 @@ def service_context(self) -> ServiceContext:
 
     def _parse_response_to_sql(self, response: str) -> str:
         """Parse response to SQL."""
-        # Find and remove SQLResult part
-        sql_result_start = response.find("SQLResult:")
         sql_query_start = response.find("SQLQuery:")
-        if sql_result_start != -1 and sql_query_start != -1:
-            response = response[sql_query_start + 1 : sql_result_start].lstrip(
```
Comment (jamesbraza) [inline llama_index/indices/struct_store/sql_query.py:250]: @logan-markewich actually someone talked about this in a PyCon talk this year.  Basically `lstrip` ([docs](https://docs.python.org/3/library/stdtypes.html#str.lstrip)) removes chars, not words

Comment (logan-markewich) [reply llama_index/indices/struct_store/sql_query.py:250]: huh, I always thought it meant left-strip? That explains a lot. Good catch!!

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Fix async streaming with azure (#7856)
 - Avoid `NotImplementedError()` in sub question generator (#7855)
 - Patch predibase initialization (#7859)
+- Simplify NL SQL response to SQL parsing, with expanded NL SQL prompt
```
Comment (logan-markewich) [inline CHANGELOG.md:11]: Can you quickly add `(#7868)` here? :)

Comment (logan-markewich) [reply CHANGELOG.md:11]: Actually, I can add that, one sec

File: llama_index/indices/struct_store/sql_query.py (hunk 1 above)
Comment (logan-markewich) [reply llama_index/indices/struct_store/sql_query.py:250]: Explains the recent errors I've seen with open-source models that are less likely to follow instructions

File: CHANGELOG.md (hunk 2 above)
Comment (jamesbraza) [reply CHANGELOG.md]: Oh thanks for adding that!

File: llama_index/indices/struct_store/sql_query.py (hunk 1 above)
Comment (jamesbraza) [reply llama_index/indices/struct_store/sql_query.py:250]: Yeah it has bitten me too before.  Thanks for the PR reviews!

**Pull request merged**